### PR TITLE
Move SetSpecialPhysicsCuts API from FairDetector to FairModule

### PR DESCRIPTION
**Pull request by sawenzel**
This promotes the SetSpecialPhysicsCuts interface to FairModule,
so that cuts can be adjusted for passive modules.

Correspoding changes were done in FairMCApplication.

### CODE REVIEW

File: base/sim/FairMCApplication.cxx [hunk 1]
```diff
@@ -1028,20 +1028,33 @@ void FairMCApplication::InitGeometry()
...
+  fModIter->Reset();
+  while((obj=fModIter->Next())) {
+    detector=dynamic_cast<FairDetector*>(obj);
+    module=dynamic_cast<FairModule*>(obj);
+    // check first if we have a FairDetector
+    if(detector) {
+      // check whether detector is active 
+      if(detector->IsActive()) {
+        detector->Initialize();
+        detector->SetSpecialPhysicsCuts();
```
[the remark this answers is not in the document]
Comment (sawenzel) [reply base/sim/FairMCApplication.cxx]: I will modify it. Thanks.

**Comment by sawenzel** [comment]
@MohammadAlTurany : I changed this PR as discussed. You should squash the commits into a single one.

**Comment by dennisklein** [comment]
@MohammadAlTurany The failing tests are unrelated to this PR - I was testing system installations of libzmq on the same machine which seems to have interfered with the CI env..

**Comment by dennisklein** [comment]
Repaired the environment and reran the tests. Looks better now, only `ex9a_static` is failing which is the usual one that fails sometimes - most likely unrelated.

**Comment by dennisklein** [comment]
@sawenzel @MohammadAlTurany Assuming the order of calling `SetSpecialPhysicsCuts()` is independent from `Initialize()/Register()` and just looking at the diff here only, I see just these two simple loops:

```cpp
for(std::list<FairDetector*>::iterator activeDetectorIter = listActiveDetectors.begin(); activeDetectorIter != listActiveDetectors.end(); activeDetectorIter++) {
  (*activeDetectorIter)->Initialize();
  (*activeDetectorIter)->Register();
}

FairModule* module;
TObject* obj;
fModIter->Reset();
while(obj = fModIter->Next()) {
  module = static_cast<FairModule*>(obj);
  assert(dynamic_cast<FairModule*>(obj));
  module->SetSpecialPhysicsCuts();
}
fModIter->Reset();
```

**Comment by MohammadAlTurany** [comment]
@dennisklein  @sawenzel We should  make a separate PR with this suggestion, I will merge now with out it.

**Pull request merged**